chatbot.py

```diff
@@ -174,16 +176,23 @@ def message_to_bot(H, clf, learn_response):
                 learn_response = LearnResponse.ORIGIN.name
         if len(proper_nouns) == 1:
             location = proper_nouns.pop()
-            if subj[0] == "geocoding" or subj[0] == location:
+            if len(subj) != 0 and (subj[0] == "geocoding" or subj[0] == location):
                 B = googleMapsApiModule.geocoding(location)
                 learn_response = LearnResponse.MESSAGE.name
-        if (len(subj) != 0 and subj[0] in ["elevation", "height", "depth"]) or (
-            len(adj) != 0 and adj[0] in ["high"]
+        if any(sub in ["elevation", "height", "depth"] for sub in subj) or (
+            "high" in adj
```

Comment:
Changed to be able to check all elements in `subj` list with an objective to cover more test cases correctly, even when the keyword we are searching for is not the first element in `subj` list.